src/controller/python/chip/bdx/bdx-transfer.cpp

```diff
@@ -0,0 +1,206 @@
+    mAwaitingAccept = false;
+    mData.assign(data_to_send.begin(), data_to_send.end());
+    mDataCount = data_to_send.size();
+
+    TransferSession::TransferAcceptData acceptData;
+    acceptData.ControlMode  = TransferControlFlags::kReceiverDrive;
+    acceptData.MaxBlockSize = mTransfer.GetTransferBlockSize();
+    acceptData.StartOffset  = mTransfer.GetStartOffset();
+    acceptData.Length       = mTransfer.GetTransferLength();
+    return mTransfer.AcceptTransfer(acceptData);
+}
+
+CHIP_ERROR BdxTransfer::Reject()
+{
+    VerifyOrReturnError(mAwaitingAccept, CHIP_ERROR_INCORRECT_STATE);
+    mAwaitingAccept = false;
+    return mTransfer.RejectTransfer(StatusCode::kTransferFailedUnknownError);
+}
+
+void BdxTransfer::HandleTransferSessionOutput(TransferSession::OutputEvent & event)
+{
+    assertChipStackLockedByCurrentThread();
+
+    ChipLogDetail(BDX, "Received event %s", event.ToString(event.EventType));
+
+    switch (event.EventType)
+    {
+    case TransferSession::OutputEventType::kInitReceived:
+        mAwaitingAccept = true;
+        mDelegate->InitMessageReceived(this, event.transferInitData);
+        break;
+    case TransferSession::OutputEventType::kStatusReceived:
+        ChipLogError(BDX, "Received StatusReport %x", static_cast<uint16_t>(event.statusData.statusCode));
+        EndSession(CHIP_ERROR_INTERNAL);
+        break;
+    case TransferSession::OutputEventType::kInternalError:
+        EndSession(CHIP_ERROR_INTERNAL);
+        break;
+    case TransferSession::OutputEventType::kTransferTimeout:
+        EndSession(CHIP_ERROR_TIMEOUT);
+        break;
+    case TransferSession::OutputEventType::kBlockReceived:
+        if (mDelegate)
+        {
+            ByteSpan data(event.blockdata.Data, event.blockdata.Length);
+            mDelegate->DataReceived(this, data);
+            mTransfer.PrepareBlockAck();
+        }
+        break;
+    case TransferSession::OutputEventType::kMsgToSend:
+        SendMessage(event);
+        if (event.msgTypeData.HasMessageType(MessageType::BlockAckEOF))
+        {
+            // TODO: Ending the session here means the StandaloneAck for the BlockAckEOF message hasn't been received.
+            EndSession(CHIP_NO_ERROR);
+        }
+        break;
+    case TransferSession::OutputEventType::kAckEOFReceived:
+        EndSession(CHIP_NO_ERROR);
+        break;
+    case TransferSession::OutputEventType::kQueryWithSkipReceived:
+        mDataTransferredCount = std::min<size_t>(mDataTransferredCount + event.bytesToSkip.BytesToSkip, mDataCount);
+        SendBlock();
+        break;
+    case TransferSession::OutputEventType::kQueryReceived:
+        SendBlock();
+        break;
+    case TransferSession::OutputEventType::kAckReceived:
+    case TransferSession::OutputEventType::kAcceptReceived:
+    case TransferSession::OutputEventType::kNone:
+        // Nothing to do.
+        break;
+    default:
+        // Should never happen.
```

Comment:
Can you add an error message here please so in case this does happen that we'll know what's up?